db/submission_test.go

```diff
@@ -59,3 +62,72 @@ func (this *DBTests) DBTestRemoveSubmission(test *testing.T) {
         }
     }
 }
+
+func (this *DBTests) DBTestFetchAttempts(test *testing.T) {
+    ResetForTesting();
+    defer ResetForTesting();
+
+    assignment := MustGetTestAssignment();
+
+    // Fetch attempts from another user who has submissions.
+    studentAttempts, err := GetSubmissionAttempts(assignment, "student@test.com");
+    if err != nil {
+        test.Fatalf("Failed to get student attempts when should be a success: %v", err);
+    }
+
+    if len(studentAttempts) == 0 {
+        test.Fatalf("Got an empty result when shouldn't be.");
+    }
+
+    // Fetch attempts from self (grader). User does not have submissions.
+    graderAttempts, err := GetSubmissionAttempts(assignment, "grader@test.com");
+    if err != nil {
+        test.Fatalf("Failed to get grader attempts when should be a success (with empty result): %v", err);
+    }
+
+    if len(graderAttempts) != 0 {
+        test.Fatalf("Got a non-empty result when should be empty.");
+    }
+
+    // User makes a submission, then that submission is removed.
+    // This case tests GetSubmissionAttempts() when the submission directory exists but is empty.
+    graderSubmission := studentAttempts[0];
+    graderSubmission.Info.User = "grader@test.com";
+
+    err = SaveSubmission(assignment, graderSubmission);
+    if err != nil {
+        test.Fatalf("Failed to save grader submission %v: ", err);
+    }
+
+    graderAttempts, err = GetSubmissionAttempts(assignment, "grader@test.com");
+    if err != nil {
+        test.Fatalf("Failed to get grader attempts when there should be one %v: ", err);
+    }
+
+    if len(graderAttempts) != 1 {
+        test.Fatalf("Fetch returned unexpected number of attempts. Expected: %d, actual: %d", 1, len(graderAttempts));
```

Comment:
s/actual/Actual/

Also, put a period at the end of the string.

(This is just a super nit, but since we are making other changes may as well make this one too.)